neurom/features/morphology.py

```diff
@@ -176,20 +223,83 @@ def trunk_angles(morph, neurite_type=NeuriteType.all):
+                                     for i in vectors[:, sort_coords]]))
 
-    # Sorting angles according to x-y plane
-    order = np.argsort(np.array([_sort_angle(i / np.linalg.norm(i), [0, 1])
-                                 for i in vectors[:, 0:2]]))
+        vectors = vectors[order]
 
-    ordered_vectors = vectors[order][:, [COLS.X, COLS.Y]]
+    # Select coordinates to consider
+    if coords_only:
+        coords = _str_to_plane(coords_only)
+        vectors = vectors[:, coords]
+
+    # Compute angles between each trunk and the next ones
+    cycling_vectors = np.vstack([vectors, vectors])
+    angles = [
+        (num_i, [
+            morphmath.angle_between_vectors(i, j)
+            for num_j, j in enumerate(cycling_vectors[num_i: num_i + len(vectors)])
+        ])
+        for num_i, i in enumerate(vectors)
+    ]
+
+    if consecutive_only:
+        angles = [sorted(i[1])[:2][-1] for i in angles if i[1]]
+    else:
+        angles = [i[1] for i in angles]
+
+    return angles
+
+
+@feature(shape=(...,))
+def trunk_angles_inter_types(
+    morph,
+    source_neurite_type=NeuriteType.apical_dendrite,
+    target_neurite_type=NeuriteType.basal_dendrite,
+    closest_component=None,
+):
+    """Calculate the angles between the trunks of the morph of a given type to another type.
+
+    For each couple of neurite, an array with 3 elements is returned:
+    * the absolute 3d angle between the two vectors.
+    * the phi angle between the two vectors.
+    * the theta angle between the two vectors.
+
+    If ``closest_component`` is set not ``None``, only one element is returned for each neurite of
+    source type:
+    * if set to 0, the one with the lowest absolute 3d angle is returned.
+    * if set to 1, the one with the lowest absolute phi angle is returned.
+    * if set to 2, the one with the lowest absolute theta angle is returned.
+    """
+    source_vectors = np.array(trunk_vectors(morph, neurite_type=source_neurite_type))
+    target_vectors = np.array(trunk_vectors(morph, neurite_type=target_neurite_type))
+
+    # In order to avoid the failure of the process in case the neurite_type does not exist
+    if len(source_vectors) == 0 or len(target_vectors) == 0:
+        return []
 
-    return [morphmath.angle_between_vectors(ordered_vectors[i], ordered_vectors[i - 1])
-            for i, _ in enumerate(ordered_vectors)]
+    angles = [
+        np.vstack([
+            np.concatenate(
+                [
+                    [morphmath.angle_between_vectors(i, j)],
+                    _spherical_from_vector(j) - _spherical_from_vector(i)
+                ]
+            )
+            for j in target_vectors
+        ])
+        for i in source_vectors
+    ]
+
+    angles = _angles_to_pi_interval(angles)
+
+    if closest_component is not None:
+        angles = [i[[np.argmin(np.abs(i[:, closest_component]))]] for i in angles if len(i) > 0]
+
+    return [i.tolist() for i in angles]
```

Comment:
I think you can just do `angles.tolist()` because it applies recursively. Please try and check https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tolist.html.

Reply:
Ah yes, cool!